.github/workflows/scripts/check_docstrings.py

```
@@ -0,0 +1,773 @@
+                # Identify violations in the file
+                violations = validate_docstring(file_path)
+
+                # Add any found violations
+                if violations:
+                    all_violations[file_path] = violations
+
+    # Return
+    return all_violations
+
+
+def main():
```

Ruff (0.12.2)

Missing return type annotation for public function `main`

Add return type annotation: `None`

(ANN201)